### ISSUE TITLE
Associated fields in Enumerator

### ISSUE THREAD

**Issue by bernardnormier** [icerpc/slicec#682]
The associated fields of an Enumerator are an:
```rs
 pub associated_fields: Option<Vec<WeakPtr<Field>>>,
```

And apparently, when slicec parses an enumerator, it:
 - parses `E()` as a non-null empty vector of fields
 - parses `E` as a null vector of fields

This seems overcomplicated. Why not simply:
```rs
 pub associated_fields: Vec<WeakPtr<Field>>,
```

with an empty vec in the common situation where the enumerator has no associated field.

**Comment by InsertCreativityHere** [comment]
This change was implemented in #685

**Comment by bernardnormier** [comment]
This change meaning:
> Changes fields to return a Vec<Field> instead of an Option<Vec<Field>> for easier use (None => vec![]).
